cdap-api/src/main/java/co/cask/cdap/api/plugin/Plugin.java

```diff
@@ -18,24 +18,44 @@
 
 import co.cask.cdap.api.artifact.ArtifactId;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
 import java.util.Objects;
 
 /**
  * A container class for holding plugin information.
  */
 public final class Plugin {
+  private final List<ArtifactId> parents;
   private final ArtifactId artifactId;
   private final PluginClass pluginClass;
   private final PluginProperties properties;
 
   public Plugin(ArtifactId artifactId, PluginClass pluginClass, PluginProperties properties) {
```

Comment:
We probably should mark this method as `@deprecated` since no one is supposed to call this anymore.